<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>2. Identify prox-eQTLs from scratch â€¢ ProxSeek</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="2. Identify prox-eQTLs from scratch">
<meta property="og:description" content="ProxSeek">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ProxSeek</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Prioritize-cis-eqtl.html">1. Prioritize cis-eQTL using ProxSeek</a>
    </li>
    <li>
      <a href="../articles/Run-Prox-eQTLs.html">2. Identify prox-eQTLs from scratch</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>2. Identify prox-eQTLs from scratch</h1>
            
      
      
      <div class="hidden name"><code>Run-Prox-eQTLs.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="prepare-input-files">1. Prepare input files<a class="anchor" aria-label="anchor" href="#prepare-input-files"></a>
</h2>
<div class="section level4">
<h4 id="since-we-are-using-matrixeqtl-to-estimate-the-association-between-genotype-and-gene-epxresion-level-the-input-file-requires-5-types--the-format-is-also-consistent-of-the-matrixeqtl-package--here-is-using-demo-data-to-show-what-the-input-file-format-should-look-like-">Since we are using MatrixEQTL to estimate the association between
genotype and gene epxresion level, the input file requires 5 types. The
format is also consistent of the MatrixEQTL package. Here is using demo
data to show what the input file format should look like.<a class="anchor" aria-label="anchor" href="#since-we-are-using-matrixeqtl-to-estimate-the-association-between-genotype-and-gene-epxresion-level-the-input-file-requires-5-types--the-format-is-also-consistent-of-the-matrixeqtl-package--here-is-using-demo-data-to-show-what-the-input-file-format-should-look-like-"></a>
</h4>
</div>
<div class="section level3">
<h3 id="first-file-snp-genotype-file">1.1 First file: SNP genotype file<a class="anchor" aria-label="anchor" href="#first-file-snp-genotype-file"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ProxSeek</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="va">SNP_file_name</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"chr19_geno2.csv"</span>, package <span class="op">=</span> <span class="st">"ProxSeek"</span><span class="op">)</span></span>
<span><span class="va">snp_file</span><span class="op">&lt;-</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">SNP_file_name</span><span class="op">)</span></span>
<span><span class="va">snp_file</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##           id 1_1 2_2 3_3 4_4</span></span>
<span><span class="co">## 1: 19:253155   1   0   0   0</span></span>
<span><span class="co">## 2: 19:253938   1   1   1   0</span></span>
<span><span class="co">## 3: 19:254304   1   1   1   0</span></span>
<span><span class="co">## 4: 19:254448   1   1   1   0</span></span>
<span><span class="co">## 5: 19:254899   1   1   1   0</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="second-file-the-genome-location-file-based-on-hg19">1.2 Second file: the genome location file, based on hg19<a class="anchor" aria-label="anchor" href="#second-file-the-genome-location-file-based-on-hg19"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snps_location_file_name</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"chr19_geno_loc2.csv"</span>, package <span class="op">=</span> <span class="st">"ProxSeek"</span><span class="op">)</span></span>
<span><span class="va">snp_location_file</span><span class="op">&lt;-</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">snps_location_file_name</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">snp_location_file</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          snp chr    pos</span></span>
<span><span class="co">## 1: 19:253155  19 253155</span></span>
<span><span class="co">## 2: 19:253938  19 253938</span></span>
<span><span class="co">## 3: 19:254304  19 254304</span></span>
<span><span class="co">## 4: 19:254448  19 254448</span></span>
<span><span class="co">## 5: 19:254899  19 254899</span></span>
<span><span class="co">## 6: 19:254927  19 254927</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="third-file-gene-expression-from-samples">1.3 Third file: gene expression from samples<a class="anchor" aria-label="anchor" href="#third-file-gene-expression-from-samples"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expression_file_name</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GE2.csv"</span>, package <span class="op">=</span> <span class="st">"ProxSeek"</span><span class="op">)</span></span>
<span><span class="va">expression_file</span><span class="op">&lt;-</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">expression_file_name</span><span class="op">)</span></span>
<span><span class="va">expression_file</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##         id        1_1        2_2        3_3         4_4</span></span>
<span><span class="co">## 1:  PPAP2C 0.00000000 0.00000000 0.00000000 0.000000000</span></span>
<span><span class="co">## 2:   MIER2 0.00801599 0.02256427 0.02540431 0.009460315</span></span>
<span><span class="co">## 3: MADCAM1 0.00000000 0.00000000 0.03838079 0.003898198</span></span>
<span><span class="co">## 4:   TPGS1 0.29582400 0.24688230 0.28883650 0.297556900</span></span>
<span><span class="co">## 5:   CDC34 0.06841568 0.10084160 0.09059088 0.040504430</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="fourth-file-gene-genomic-location-file-hg19-as-reference">1.4 Fourth file: gene genomic location file, hg19 as reference<a class="anchor" aria-label="anchor" href="#fourth-file-gene-genomic-location-file-hg19-as-reference"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_location_file_name</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"geneloc_TSS2.csv"</span>, package <span class="op">=</span> <span class="st">"ProxSeek"</span><span class="op">)</span></span>
<span><span class="va">gene_location</span><span class="op">&lt;-</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">gene_location_file_name</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_location</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     geneid chr     s1     s2</span></span>
<span><span class="co">## 1:  PPAP2C  19 281040 281040</span></span>
<span><span class="co">## 2:   MIER2  19 305575 305575</span></span>
<span><span class="co">## 3: MADCAM1  19 489176 489176</span></span>
<span><span class="co">## 4:   TPGS1  19 507299 507299</span></span>
<span><span class="co">## 5:   CDC34  19 531714 531714</span></span>
<span><span class="co">## 6:    GZMM  19 544034 544034</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="last-file-the-covariants-file">1.5 Last file: the covariants file<a class="anchor" aria-label="anchor" href="#last-file-the-covariants-file"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covariates_file_name</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"Covariates.csv"</span>, package <span class="op">=</span> <span class="st">"ProxSeek"</span><span class="op">)</span></span>
<span><span class="va">covariates_file</span><span class="op">&lt;-</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">covariates_file_name</span><span class="op">)</span></span>
<span><span class="va">covariates_file</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##     id         1_1         2_2        3_3         4_4</span></span>
<span><span class="co">## 1: pc1  0.00217631  0.01010850 0.01684490 -0.06331710</span></span>
<span><span class="co">## 2: pc2  0.02619680  0.00232193 0.00380629 -0.04022280</span></span>
<span><span class="co">## 3: pc3  0.03116540 -0.02786010 0.00959531 -0.01843700</span></span>
<span><span class="co">## 4: pc4 -0.04101080 -0.02468170 0.03026220 -0.01714570</span></span>
<span><span class="co">## 5: pc5 -0.00673392 -0.04362350 0.02123780 -0.00509033</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="now-we-can-run-proxseek-to-identify-prox-eqtls">Now we can run ProxSeek to identify prox-eQTLs<a class="anchor" aria-label="anchor" href="#now-we-can-run-proxseek-to-identify-prox-eqtls"></a>
</h2>
<pre><code><span><span class="co">## [1] "/tmp/Rtmpkvr3pa/temp_libpath74e77fc2c8f6/ProxSeek/data"</span></span></code></pre>
<pre><code>## P-PIR file reported 25511 genes!Shared number of genes with cis-eQTL report 49
  |                                                                            
  |                                                                      |   0%progress will be called as each result is returned
## discovered package(s): 
## automatically exporting the following variables from the local environment:
##   cell.hic, eqtl.df.sub, kk.new 
## explicitly exporting package(s): 
## numValues: 11, numResults: 1, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=                                                                     |   2%numValues: 12, numResults: 2, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |===                                                                   |   4%numValues: 13, numResults: 3, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |====                                                                  |   6%numValues: 14, numResults: 4, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |======                                                                |   8%numValues: 15, numResults: 5, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=======                                                               |  10%numValues: 16, numResults: 6, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=========                                                             |  12%numValues: 17, numResults: 7, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |==========                                                            |  14%numValues: 18, numResults: 8, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |===========                                                           |  16%numValues: 19, numResults: 9, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=============                                                         |  18%numValues: 20, numResults: 10, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |==============                                                        |  20%numValues: 21, numResults: 11, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |================                                                      |  22%numValues: 22, numResults: 12, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=================                                                     |  24%numValues: 23, numResults: 13, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |===================                                                   |  27%numValues: 24, numResults: 14, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |====================                                                  |  29%numValues: 25, numResults: 15, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=====================                                                 |  31%numValues: 26, numResults: 16, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=======================                                               |  33%numValues: 27, numResults: 17, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |========================                                              |  35%numValues: 28, numResults: 18, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |==========================                                            |  37%numValues: 29, numResults: 19, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |===========================                                           |  39%numValues: 30, numResults: 20, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=============================                                         |  41%numValues: 31, numResults: 21, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |==============================                                        |  43%numValues: 32, numResults: 22, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |===============================                                       |  45%numValues: 33, numResults: 23, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=================================                                     |  47%numValues: 34, numResults: 24, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |==================================                                    |  49%numValues: 35, numResults: 25, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |====================================                                  |  51%numValues: 36, numResults: 26, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=====================================                                 |  53%numValues: 37, numResults: 27, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=======================================                               |  55%numValues: 38, numResults: 28, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |========================================                              |  57%numValues: 39, numResults: 29, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=========================================                             |  59%numValues: 40, numResults: 30, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |===========================================                           |  61%numValues: 41, numResults: 31, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |============================================                          |  63%numValues: 42, numResults: 32, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |==============================================                        |  65%numValues: 43, numResults: 33, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |===============================================                       |  67%numValues: 44, numResults: 34, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=================================================                     |  69%numValues: 45, numResults: 35, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |==================================================                    |  71%numValues: 46, numResults: 36, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |===================================================                   |  73%numValues: 47, numResults: 37, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |=====================================================                 |  76%numValues: 48, numResults: 38, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |======================================================                |  78%numValues: 49, numResults: 39, stopped: FALSE
## returning status FALSE
## 
  |                                                                            
  |========================================================              |  80%numValues: 49, numResults: 39, stopped: TRUE
## numValues: 49, numResults: 40, stopped: TRUE
## returning status FALSE
## 
  |                                                                            
  |=========================================================             |  82%numValues: 49, numResults: 41, stopped: TRUE
## returning status FALSE
## 
  |                                                                            
  |===========================================================           |  84%numValues: 49, numResults: 42, stopped: TRUE
## returning status FALSE
## 
  |                                                                            
  |============================================================          |  86%numValues: 49, numResults: 43, stopped: TRUE
## returning status FALSE
## 
  |                                                                            
  |=============================================================         |  88%numValues: 49, numResults: 44, stopped: TRUE
## returning status FALSE
## 
  |                                                                            
  |===============================================================       |  90%numValues: 49, numResults: 45, stopped: TRUE
## returning status FALSE
## 
  |                                                                            
  |================================================================      |  92%numValues: 49, numResults: 46, stopped: TRUE
## returning status FALSE
## 
  |                                                                            
  |==================================================================    |  94%numValues: 49, numResults: 47, stopped: TRUE
## returning status FALSE
## 
  |                                                                            
  |===================================================================   |  96%numValues: 49, numResults: 48, stopped: TRUE
## returning status FALSE
## 
  |                                                                            
  |===================================================================== |  98%numValues: 49, numResults: 49, stopped: TRUE
## first call to combine function
## evaluating call object to combine results:
##   fun(result.1, result.2, result.3, result.4, result.5, result.6, 
##     result.7, result.8, result.9, result.10, result.11, result.12, 
##     result.13, result.14, result.15, result.16, result.17, result.18, 
##     result.19, result.20, result.21, result.22, result.23, result.24, 
##     result.25, result.26, result.27, result.28, result.29, result.30, 
##     result.31, result.32, result.33, result.34, result.35, result.36, 
##     result.37, result.38, result.39, result.40, result.41, result.42, 
##     result.43, result.44, result.45, result.46, result.47, result.48, 
##     result.49)
## returning status TRUE
## 
  |                                                                            
  |======================================================================| 100%</code></pre>
<pre><code><span><span class="co">##         snps  gene statistic       pvalue          FDR       beta chr    pos</span></span>
<span><span class="co">## 1: 19:890165 MED16  20.64533 9.082635e-79 9.767258e-72 0.04529877  19 890165</span></span>
<span><span class="co">## 2: 19:882724 MED16  20.56403 2.918949e-78 1.569485e-71 0.04519645  19 882724</span></span>
<span><span class="co">## 3: 19:874647 MED16  20.48710 8.797387e-78 1.665871e-71 0.04549324  19 874647</span></span>
<span><span class="co">## 4: 19:874658 MED16  20.48710 8.797387e-78 1.665871e-71 0.04549324  19 874658</span></span>
<span><span class="co">## 5: 19:875389 MED16  20.46886 1.142450e-77 1.665871e-71 0.04547395  19 875389</span></span>
<span><span class="co">## 6: 19:875902 MED16  20.46886 1.142450e-77 1.665871e-71 0.04547395  19 875902</span></span>
<span><span class="co">##    Symbol   Within</span></span>
<span><span class="co">## 1:  MED16 Promoter</span></span>
<span><span class="co">## 2:  MED16 Promoter</span></span>
<span><span class="co">## 3:  MED16 Promoter</span></span>
<span><span class="co">## 4:  MED16 Promoter</span></span>
<span><span class="co">## 5:  MED16 Promoter</span></span>
<span><span class="co">## 6:  MED16 Promoter</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Xinge Wang.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
